ansible_collections/arista/avd/roles/eos_designs/python_modules/overlay/router_bgp.py

```diff
@@ -45,11 +45,12 @@ def _bgp_cluster_id(self) -> str | None:
                 return get(self._hostvars, "switch.bgp_cluster_id", default=self._router_id)
         return None
 
-    def _generate_base_peer_group(self, pg_type: str, pg_name: str) -> dict:
+    def _generate_base_peer_group(self, peer_group_name: str, pg_type: str, pg_name: str) -> dict:
         if pg_type not in ["mpls", "evpn"]:
             raise AristaAvdError("_generate_base_peer_group should be called with pg_type in ['mpls', 'evpn']")
 
         peer_group = {
+            "name": peer_group_name,
```

Comment:
We could set this name similar to how we set password etc below. So
```suggestion
            "name": get(self._hostvars, f"switch.bgp_peer_groups.{pg_name}.name"),
```